denops/@ddu-sources/git_log.ts

```diff
@@ -9,27 +9,94 @@ import { ErrorStream } from "../ddu-source-git_log/message.ts";
 
 type Params = {
   cwd?: string;
+  isGraph?: boolean;
+  isAll?: boolean;
+  isReverse?: boolean;
+  commitOrder?:
+    | "--date-order"
+    | "--author-date-order"
+    | "--topo-order";
 };
 
-function formatLog(): string {
-  return [
+function formatLog(isGraph: boolean): string {
+  const baseFormat = [
     "%H", // Hash
     "%aN", // Author
     "%ai", // AuthorDate
     "%cN", // Commit
     "%ci", // CommitDate
     "%s", // Subject
-  ].join("%x00");
+  ];
+  if (isGraph) return ["", ...baseFormat].join("%x00");
+  return baseFormat.join("%x00");
 }
 
-function parseLog(cwd: string, line: string): Item<ActionData> {
-  const [hash, author, authDate, committer, commitDate, subject] = line.split(
-    "\x00",
-  );
+function parseLog(
+  cwd: string,
+  line: string,
+  isGraph: boolean,
+): Item<ActionData> {
+  let graph!: string;
+  let hash: string;
+  let author: string;
+  let authDate: string;
+  let committer: string;
+  let commitDate: string;
+  let subject: string;
+
+  if (isGraph) {
+    [
+      graph,
+      hash,
+      author,
+      authDate,
+      committer,
+      commitDate,
+      subject,
+    ] = line.split("\x00");
+  } else {
+    [
+      hash,
+      author,
+      authDate,
+      committer,
+      commitDate,
+      subject,
+    ] = line.split("\x00");
+  }
```

Comment:
```suggestion
  [
    graph,
    hash,
    author,
    authDate,
    committer,
    commitDate,
    subject,
  ] = line.split("\x00");
```

ref: https://github.com/kyoh86/ddu-source-git_log/pull/3/files#r1285952469